### DISCUSSION TITLE
Connect to Azure Service Bus using RabbitMQ.Client

### DISCUSSION THREAD

**Discussion by hence-proven** [rabbitmq/rabbitmq-dotnet-client#1353] [Q&A] [answered]
I am migrating set of services to Azure cloud environment and need to publish and consume messages, as message broker is selected Azure Service Bus, so I am trying to migrate with minimal changes.

From Service Bus docs, I found how to convert SAS Policy Connection string to AMQP Connection String (https://red-mushroom-0f7446a0f.azurestaticapps.net/)

```
using RabbitMQ.Client;

var factory = new ConnectionFactory
{
    Uri = new Uri("amqps://<policy_name>:<policy_primary_key>@<service_bus_name>.servicebus.windows.net:5671/?sasl=plain")
};
;
var client = factory.CreateConnection().CreateModel();

```
and on creation connection / model I'm getting error 
```
RabbitMQ.Client.Exceptions.BrokerUnreachableException: 'None of the specified endpoints were reachable'
----
This exception was originally thrown at this call stack:
    RabbitMQ.Client.Framing.Impl.Connection.StartAndTune() in Connection.cs
    RabbitMQ.Client.Framing.Impl.Connection.Open(bool) in Connection.cs
    RabbitMQ.Client.Framing.Impl.Connection.Connection(RabbitMQ.Client.IConnectionFactory, bool, RabbitMQ.Client.Impl.IFrameHandler, string) in Connection.cs
    RabbitMQ.Client.Framing.Impl.Connection.Connection(RabbitMQ.Client.IConnectionFactory, bool, RabbitMQ.Client.Impl.IFrameHandler, System.Buffers.ArrayPool<byte>, string) in Connection.cs
    RabbitMQ.Client.Framing.Impl.AutorecoveringConnection.Init(RabbitMQ.Client.Impl.IFrameHandler) in AutorecoveringConnection.cs
    RabbitMQ.Client.Framing.Impl.AutorecoveringConnection.Init(RabbitMQ.Client.IEndpointResolver) in AutorecoveringConnection.cs
    RabbitMQ.Client.ConnectionFactory.CreateConnection(RabbitMQ.Client.IEndpointResolver, string) in ConnectionFactory.cs
```

So the question, is the client not properly configured? Or the client library is nor supposed to use in such case?

**Comment by lukebakken** [comment]
`None of the specified endpoints were reachable` is pretty clear - a TCP connection can't be made. Please see these guides:
* https://www.rabbitmq.com/troubleshooting-networking.html
* https://www.rabbitmq.com/troubleshooting-ssl.html

Start with `openssl s_client` to ensure a TLS connection can be made. If you'd like further assistance, please tell us exactly what you try and provide a full transcript of the command(s) and their output.

**Comment by michaelklishin** [comment, marked as answer]
This client implements AMQP 0-9-1, Azure Service Bus only implements AMQP 1.0 (a completely different protocols despite the name).

Use an AMQP 1.0 client or use this client with RabbitMQ (which supports AMQP 0-9-1, AMQP 1.0, RabbitMQ Stream Protocol, MQTTv3 and soon MQTTv5, STOMP)

**Reply by lukebakken** [reply]
> Azure Service Bus only implements AMQP 1.0

I didn't know that! Now I do. Thanks @michaelklishin
